openmp/libomptarget/plugins-nextgen/common/src/GlobalHandler.cpp

```diff
@@ -53,9 +57,15 @@ Error GenericGlobalHandlerTy::getGlobalMetadataFromELF(
     const ELF64LE::Shdr &Section, GlobalTy &ImageGlobal) {
 
   // The global's address is computed as the image begin + the ELF section
-  // offset + the ELF symbol value.
-  ImageGlobal.setPtr(advanceVoidPtr(
-      Image.getStart(), Section.sh_offset - Section.sh_addr + Symbol.st_value));
+  // offset + the ELF symbol value except for NOBITS sections that, as the name
+  // suggests, have no bits in the image. We still record the size and use
+  // nullptr to indicate there is no location.
+  if (Section.sh_type == ELF::SHT_NOBITS)
```

Comment:
Not directly related, but could we remove this function from `GenericGlobalHandlerTy` and just make this a utility in `Elf.cpp` like `getSymbolAddr` that returns said void pointer?

Reply:
Sure, it makes more sense to avoid propagating ELF specific stuff from the generic interface.